### PR TITLE
feat: allow setting db UUID

### PR DESCRIPTION
**Pull request by betodealmeida**
### SUMMARY

Allow client to specify an UUID when creating a database.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TESTING INSTRUCTIONS

I added a unit test. I also refactored some logic into a `full_api_access` fixture that helps write integration tests.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/databases/schemas.py [hunk 1]
```diff
@@ -397,6 +397,7 @@ class Meta:  # pylint: disable=too-few-public-methods
     )
     is_managed_externally = fields.Boolean(allow_none=True, default=False)
     external_url = fields.String(allow_none=True)
+    uuid = fields.String(allow_none=True)
```
Comment (zephyring) [inline superset/databases/schemas.py:400]: should we default to some generated uuid if not provided? I think making it allow_none=False will be more helpful.

Comment (zephyring) [reply superset/databases/schemas.py:400]: we might need some migration to backfill though

Comment (betodealmeida) [reply superset/databases/schemas.py:400]: Right, this allows the user to pass `None` for the UUID, and in that case a random one will be generated.

Comment (betodealmeida) [reply superset/databases/schemas.py]: @zephyring I changed it so that it's not required, but if present it can't be `None`, which is what we want.

**Review by zephyring — APPROVED** [review]

**Review by eschutho — APPROVED** [review]

**Comment by eschutho** [comment]
@betodealmeida does this need a migration or is it just an in-memory sqlalchemy construct?

**Comment by betodealmeida** [comment]
> @betodealmeida does this need a migration or is it just an in-memory sqlalchemy construct?

No migration needed, we're just changing the payload schema to allow an optional UUID.

**Pull request merged**